app/services/articles/page_view_updater.rb

```diff
@@ -0,0 +1,33 @@
+module Articles
+  # This module is responsible for updating a specific page view for a given article and user.
+  #
+  # @see Articles::UpdatePageViewsWorker for the sibling that's responsible for recording page
+  #      views.
+  module PageViewUpdater
+    # @param article_id [Integer]
+    # @param user_id [Integer]
+    #
+    # @return [TrueClass] if we updated or created a PageView
+    # @return [FalseClass] if we did not update a PageView
+    #
+    # @note Regardless of return status, consider the business logic successful unless we raise an
+    #       exception.  The return value is present for easing testing.  The `find_or_create_by`
+    #       adds a complication in the testing logic
+    #       (e.g., `expect { Articles::PageViewUpdater.call }.not_to change(PageView, :count) `
+    def self.call(article_id:, user_id:)
+      # Don't record views to unpublished articles.
+      return false if Article.unpublished.exists?(id: article_id)
+      # Don't record author's own views.
+      return false if Article.published.exists?(id: article_id, user_id: user_id)
```

Comment:
Nice solution to this! Avoiding instantiating the object or even pulling back the full `articles` row! 💯

Reply:
I can do it in one query if we want.  It just looked uglier by a lot.